GVRf/Framework/jni/objects/textures/cubemap_texture.h

```diff
@@ -0,0 +1,93 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/***************************************************************************
+ * Cube map texture made by six bitmaps.
+ ***************************************************************************/
+
+#ifndef CUBEMAP_TEXTURE_H_
+#define CUBEMAP_TEXTURE_H_
+
+#include <string>
+
+#include <android/bitmap.h>
+
+#include "objects/textures/texture.h"
+#include "util/gvr_log.h"
+
+namespace gvr {
+class CubemapTexture: public Texture {
+public:
+	explicit CubemapTexture(JNIEnv* env, jobjectArray bitmapArray) :
```

Comment:
Everywhere else, the `_jni.cpp` files contain all the JNI code, and make calls into the 'straight C++ code'. I think it would be best to follow that convention, here: do all the `env->` in `cubemap_texture_jni.cpp` and just pass in an array of six structs with the `AndroidBitmapInfo info` and the `void* pixels`.